dbms/src/Interpreters/ExpressionAnalyzer.cpp

```diff
@@ -2792,7 +2792,7 @@ void ExpressionAnalyzer::collectUsedColumns()
             required.insert(column_name_type.name);
 
     /// You need to read at least one column to find the number of rows.
-    if (required.empty())
+    if (select_query && required.empty())
```

Comment:
This will cause the  `Actions->getRequireColumns` to get incorrect result. Besides judging the type of ast,I can't think of a better way. Maybe you can give me some advice?